python/triton/language/core.py

```diff
@@ -312,15 +312,17 @@ def __init__(self, element_ty: dtype, shape: List):
 
         self.shape = shape
         self.numel = 1
-        for s in self.shape:
-            self.numel *= s
-        if self.numel > TRITON_MAX_TENSOR_NUMEL:
-            raise ValueError(f"numel ({self.numel}) exceeds triton maximum tensor numel ({TRITON_MAX_TENSOR_NUMEL})")
+        print('shape', shape)
+        if semantic._is_static_shape(shape):
```

Comment:
We reuse the `shpae` member within the `block_type`, so that the symbolic shape could propagate throughout the kernel without large modification to semantic.py